spec/syntax.md

```diff
@@ -705,6 +705,43 @@ on the pairing, ordering, or contents of _markup_ during _formatting_.
 Examples in this section are meant to be illustrative and
 might not match future requirements or usage.
 
+> [!NOTE]
+> The Tech Preview does not provide a built-in mechanism for overriding
+> values in the _formatting context_ (most notably the locale)
+> Nor does it provide a mechanism for identifying specific expressions
+> such as by assigning a name or id.
+> The utility of these types of mechanisms has been debated.
+> There are at least two proposed mechanisms for implementing support for
+> these. 
+> Specifically, one mechanism would be to reserve specifically-named options, 
+> possibly using a Unicode namespace (i.e. `locale=xxx` or `u:locale=xxx`).
+> Such options would be reserved for use in any and all functions or markup.
+> The other mechanism would be to use the reserved "expression attribute" syntax
+> for this purpose (i.e. `@locale=xxx` or `@id=foo`)
+> Neither mechanism was included in this Tech Preview.
+> Feedback on the preferred mechanism for managing these features
+> is strongly desired.
+> 
+> In the meantime, function authors and other implementers are cautioned to avoid creating 
+> function-specific or implementation-specific option values for this purpose.
+> One workaround would be to use the implementation's namespace for these 
+> features to insure later interoperability when such a mechanism is finalized 
+> during the Tech Preview period.
+> Specifically:
+> - Avoid specifying an option for setting the locale of an expression as different from
+>   that of the overall _message_ locale, or use a namespace that later maps to the final
+>   mechanism.
+> - Avoid specifying options for the purpose of linking placeholders
+>   (such as to pair opening markup to closing markup) 
+>   or require that such options use an implementation-specific namespace
+>   while cautioning users that such option might be replaced or unavailable later.
```

Comment:
The grammar is off here, depending on what you mean.  

```
Avoid specifying options for X
or require that such options use Y 
     while cautioning users that such option Z
```
I'm guessing you mean:
```
Avoid specifying options for X
or **requiring** that such options use Y 
     while cautioning users that such **options** Z
```

Reply:
Hah, this is exactly what I meant in my comment above: https://github.com/unicode-org/message-format-wg/pull/647#discussion_r1494754072 :)

My position is that we should recommend that early adopters "avoid or require" — and not: "avoid specifying and avoid requiring".

What would be a good way to phrase this?

Reply:
I think it should probably be:

```suggestion
> - Avoid specifying options for the purpose of linking placeholders
>   (such as to pair opening markup to closing markup).
>   If such an option is created, the implementer should use an
>   implementation-specific namespace.
>   Users and implementers are cautioned that such options might be
>   replaced with a standard mechanism in a future version.
```